src/workerd/api/tests/compat-flags-test.js

```diff
@@ -0,0 +1,50 @@
+import {
+  ok,
+  strictEqual,
+  throws,
+} from 'node:assert';
+
+const kOn = 'on';
+const kOff = 'off';
+
+const { compatibilityFlags } = globalThis.Cloudflare;
+ok(Object.isSealed(compatibilityFlags));
+ok(Object.isFrozen(compatibilityFlags));
+ok(!Object.isExtensible(compatibilityFlags));
+
+// It shuld be possible to shadow the Cloudflare global
+const Cloudflare = 1;
+
+export const compatFlagsTest = {
+  test() {
+    // The compatibility flags object should be sealed, frozen, and not extensible.
+    throws(() => compatibilityFlags.no_nodejs_compat_v2 = "...");
+    throws(() => compatibilityFlags.not_a_real_compat_flag = "...");
+    throws(() => { delete compatibilityFlags['nodejs_compat_v2']; });
+
+    // The compatibility flags object should have no prototype.
+    strictEqual(Object.getPrototypeOf(compatibilityFlags), null);
+
+    // The compatibility flags object should have the expected properties.
+    strictEqual(compatibilityFlags['nodejs_compat_v2'], kOn);
+    strictEqual(compatibilityFlags['no_nodejs_compat_v2'], kOff);
```

Comment:
I like that you added support `'no_'` variant as well - this will improve code readability in some cases.